client/state/data-layer/wpcom/me/sites/index.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * Internal dependencies
+ */
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { receiveSites } from 'state/sites/actions';
+import {
+	SITES_REQUEST,
+	SITES_REQUEST_FAILURE,
+	SITES_REQUEST_SUCCESS,
+} from 'state/action-types';
+
+/**
+ * Object holding query parameters for request all sites (exported for testing)
+ */
+export const SITES_REQUEST_QUERY_PARAMS = {
+	site_visibility: 'all',
+	include_domain_only: true,
+	site_activity: 'active',
+	fields: 'ID,URL,name,capabilities,jetpack,visible,is_private,is_vip,icon,plan,jetpack_modules,single_user_site,is_multisite,options', //eslint-disable-line max-len
```

Comment:
Yes, I totally agree I used the same approach we had, but yes that can be improved. As soon as we revisit this PR after our action plan. I will improve this and other suggestions that may arise. Thank you for the suggestion!